pkg/controller/download_request_controller_test.go

```diff
@@ -18,306 +18,249 @@ package controller
-			backup:          defaultBackup(),
-			backupLocation:  newBackupLocation("a-location", "a-provider", "a-bucket"),
-			expectGetsURL:   true,
-		},
-		{
-			name:            "restore results request with phase 'New' gets a url",
-			downloadRequest: newDownloadRequest(velerov1api.DownloadRequestPhaseNew, velerov1api.DownloadTargetKindRestoreResults, "a-backup-20170912150214"),
-			restore:         builder.ForRestore(velerov1api.DefaultNamespace, "a-backup-20170912150214").Phase(velerov1api.RestorePhaseCompleted).Backup("a-backup").Result(),
-			backup:          defaultBackup(),
-			backupLocation:  newBackupLocation("a-location", "a-provider", "a-bucket"),
-			expectGetsURL:   true,
-		},
-		{
-			name:            "request with phase 'Processed' is not deleted if not expired",
-			downloadRequest: newDownloadRequest(velerov1api.DownloadRequestPhaseProcessed, velerov1api.DownloadTargetKindBackupLog, "a-backup-20170912150214"),
-			backup:          defaultBackup(),
-		},
-		{
-			name:            "request with phase 'Processed' is deleted if expired",
-			downloadRequest: newDownloadRequest(velerov1api.DownloadRequestPhaseProcessed, velerov1api.DownloadTargetKindBackupLog, "a-backup-20170912150214"),
-			backup:          defaultBackup(),
-			expired:         true,
-		},
-	}
+	DescribeTable("a Download request",
+		func(test request) {
+			// now will be used to set the fake clock's time; capture
+			// it here so it can be referenced in the test case defs.
+			now, err := time.Parse(time.RFC1123, time.RFC1123)
+			Expect(err).To(BeNil())
+			now = now.Local()
 
-	for _, tc := range tests {
-		t.Run(tc.name, func(t *testing.T) {
-			var fakeClient client.Client
-			if tc.backupLocation != nil {
-				fakeClient = velerotest.NewFakeControllerRuntimeClient(t, tc.backupLocation)
-			} else {
-				fakeClient = velerotest.NewFakeControllerRuntimeClient(t)
-			}
+			rClock := clock.NewFakeClock(now)
 
-			harness := newDownloadRequestTestHarness(t, fakeClient)
+			const signedURLTTL = 10 * time.Minute
 
-			// set up test case data
+			var (
+				pluginManager = &pluginmocks.Manager{}
+				backupStores  = make(map[string]*persistencemocks.BackupStore)
+			)
+			pluginManager.On("CleanupClients").Return(nil)
+
+			Expect(test.downloadRequest).ToNot(BeNil())
 
 			// Set .status.expiration properly for processed requests. Since "expired" is relative to the controller's
 			// clock time, it's easier to do this here than as part of the test case definitions.
-			if tc.downloadRequest != nil && tc.downloadRequest.Status.Phase == velerov1api.DownloadRequestPhaseProcessed {
-				if tc.expired {
-					tc.downloadRequest.Status.Expiration = &metav1.Time{Time: harness.controller.clock.Now().Add(-1 * time.Minute)}
+			if test.downloadRequest != nil && test.downloadRequest.Status != (velerov1api.DownloadRequestStatus{}) && test.downloadRequest.Status.Phase == velerov1api.DownloadRequestPhaseProcessed {
```

Comment:
I think this whole outer check can be removed as we need to set the expiration for all requests.

Reply:
The expiration gets set only in the same block of code where we update the request status to `processed` (https://github.com/vmware-tanzu/velero/blob/1c536aced492862ed61842315921ab6ccfe62887/pkg/controller/download_request_controller.go#L113). 

This means, not all requests have an expiration. A request that has the phase `""` or `New` will not have its expiration set until it gets processed and returned successfully from the reconcile loop.

Reply:
Which is to say, only requests that have been successfully processed will have it's expiration set.

Reply:
Hm. You are right, of course. Nvm what I said. Let me re-evaluate these tests.